lib/client-assets.php

```diff
@@ -203,7 +203,13 @@ function gutenberg_register_packages_scripts( $scripts ) {
 		// Add dependencies that cannot be detected and generated by build tools.
 		switch ( $handle ) {
 			case 'wp-block-library':
-				array_push( $dependencies, 'editor' );
+				if (
+					! gutenberg_is_experiment_enabled( 'gutenberg-no-tinymce' ) ||
+					! empty( $_GET['requiresTinymce'] ) ||
+					gutenberg_current_content_contains_classic_block()
```

Comment:
We're also loading the classic block and the editor assets if:
* The experiment is disabled
* There is a classic block detected in the content of the current post
* The "require tinymce" query argument is set

Reply:
I think this would imply we want to do this magic with other groups of assets. Since I'm not sure this will be a necessity, I prefer keeping it simple as per the YAGNI principle and leaving this just as an idea.